de.metas.contracts/src/main/java/de/metas/flatrate/api/IContractChangeBL.java

```diff
@@ -50,7 +50,15 @@
 	 *            </ul>
 	 * @param changeDate the cancellation date. If this this date is before the term's "regular" EndDate, it is also used to find the correct {@link de.metas.flatrate.model.I_C_Contract_Change} record
 	 *            for the cancel conditions.
+	 * @param isCloseInvoiceCandidate this value is forwarded to the given term's <code>IsCloseInvoiceCandidate</code> column and will determine what to do with invoice candidates for the term which were not
+	 *            yet (fully) invoiced. See {@link de.metas.flatrate.invoicecandidate.spi.impl.FlatrateTermHandler}
 	 * @throws SubscriptionChangeException if <code>changeDate</code> is before the term's EndDate and if there is no {@link de.metas.flatrate.model.I_C_Contract_Change} record for that date
 	 */
-	void cancelContract(I_C_Flatrate_Term term, Timestamp changeDate);
+	void cancelContract(I_C_Flatrate_Term term, Timestamp changeDate,  boolean isCloseInvoiceCandidate);
+	
+	/**
+	 * ending naturally a contract
```

Comment:
When we have a limited contract, for example, a 3 months contract, when those 3 months expire, we want that the contract to have a status that is saying that: that the contract ended naturally.